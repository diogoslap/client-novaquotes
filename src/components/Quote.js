import React,{Component} from 'react'
class Quote extends Component{
  state ={
    author:'',
    quote:{}
  }

  componentDidMount(){
    const {quote} = this.props
    this.setState({
      author:quote.author,
      quote:quote
    })
  }

  render(){
    const quote = this.state.quote    
    return(
      <div className="col-lg-3 col-md-4 col-sm-6 col-xs-12 generic">
        <div className="card no-border" >
          <div className="header-bg">
          </div>
          <div className="avatar">
            {!this.state.author && <img src={quote.profile_picture} alt={quote.display_name}/>}
            {this.state.author &&  <img src={this.state.author.profile_picture} alt={this.state.author.display_name}/>}
          </div>
          <div className="card-body">
            {this.state.author &&  <p className="card-text font-weight-bold name">{this.state.author.display_name}</p>}
            {!this.state.author && <p className="card-text font-weight-bold name">{quote.display_name}</p>}            
            <p className="card-text capitalize">{quote.message}</p>            
          </div>
          {this.state.author &&
          <div className="card-footer">
            <p className="card-text name">Sent by: {quote.display_name}</p>
          </div>
          }
        </div>
      </div>
  );
  }
}

export default Quote

