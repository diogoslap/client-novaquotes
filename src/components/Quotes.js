import React,{Component} from 'react'
import Quote from './Quote'

class Quotes extends Component{
  state={
    limit:8,
    page:0,
    showButton:true
  }
  componentDidMount(){    
    this.props.loadQuotes(this.state)
    this.setState({
      page:this.state.page+1
    })
  }

  loadMore = () =>{
    this.setState({
      page:this.state.page+1
    })
    this.props.loadQuotes(this.state)
    this.showButton()
  }

  showButton = () =>{
    const {quotes} = this.props
    let total = this.state.limit * this.state.page+1
    if(quotes.length < total){
      this.setState({
        showButton:false
      }) 
    }else{
      this.setState({
        showButton:true
      }) 
    }    
  }

  render(){
    const {quotes} = this.props
    return(
      <div className="row">        
        {quotes.map((quote) => <Quote key={quote.id} quote={quote}/>)}
        <div className="col text-center">
          {this.state.showButton && <button className="btn btn-primary" onClick={this.loadMore}>Ver mais</button>}
        </div>
      </div>
    )
  }
}

export default Quotes