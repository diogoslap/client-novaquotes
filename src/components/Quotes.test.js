import React from 'react'
import {shallow} from 'enzyme'
import Quotes from './Quotes'
import Quote from './Quote'

describe('<Quotes />',()=>{   
  it("Should render Quotes", () => {
    const quotes = [
      {id:'22',message:'Teste', created_date:new Date('2019-01-01'),username:'teste',display_name:'Teste User'},
      {id:'21',message:'Teste 2', created_date:new Date('2019-01-02'),username:'teste',display_name:'Teste User 2'},
    ]
    
    const api = { 
      loadQuotes: jest.fn()
    }
    const wrapper = shallow(<Quotes loadQuotes={api.loadQuotes}  quotes={quotes}/>)
    expect(wrapper).toMatchSnapshot() 
    expect(wrapper.find(Quote).length).toBe(2)
    expect(wrapper.find(Quote).get(1).key).toBe(quotes[1].id)    
    //Get author username from quote 1
    const element = shallow(wrapper.find(Quote).get(0))
    expect(element.find('.card-body .card-text.font-weight-bold.name').text()).toEqual(quotes[0].display_name);
    expect(element.find('.card-body .card-text.capitalize').text()).toEqual(quotes[0].message);
    
  });   
})