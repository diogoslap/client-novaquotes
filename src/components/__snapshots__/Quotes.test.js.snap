// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quotes /> Should render Quotes 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Quotes
    loadQuotes={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "limit": 8,
              "page": 0,
              "showButton": true,
            },
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    quotes={
      Array [
        Object {
          "created_date": 2019-01-01T00:00:00.000Z,
          "display_name": "Teste User",
          "id": "22",
          "message": "Teste",
          "username": "teste",
        },
        Object {
          "created_date": 2019-01-02T00:00:00.000Z,
          "display_name": "Teste User 2",
          "id": "21",
          "message": "Teste 2",
          "username": "teste",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Quote
            quote={
              Object {
                "created_date": 2019-01-01T00:00:00.000Z,
                "display_name": "Teste User",
                "id": "22",
                "message": "Teste",
                "username": "teste",
              }
            }
          />,
          <Quote
            quote={
              Object {
                "created_date": 2019-01-02T00:00:00.000Z,
                "display_name": "Teste User 2",
                "id": "21",
                "message": "Teste 2",
                "username": "teste",
              }
            }
          />,
        ],
        <div
          className="col text-center"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Ver mais
          </button>
        </div>,
      ],
      "className": "row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "22",
        "nodeType": "class",
        "props": Object {
          "quote": Object {
            "created_date": 2019-01-01T00:00:00.000Z,
            "display_name": "Teste User",
            "id": "22",
            "message": "Teste",
            "username": "teste",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "21",
        "nodeType": "class",
        "props": Object {
          "quote": Object {
            "created_date": 2019-01-02T00:00:00.000Z,
            "display_name": "Teste User 2",
            "id": "21",
            "message": "Teste 2",
            "username": "teste",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            className="btn btn-primary"
            onClick={[Function]}
          >
            Ver mais
          </button>,
          "className": "col text-center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ver mais",
            "className": "btn btn-primary",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Ver mais",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Quote
              quote={
                Object {
                  "created_date": 2019-01-01T00:00:00.000Z,
                  "display_name": "Teste User",
                  "id": "22",
                  "message": "Teste",
                  "username": "teste",
                }
              }
            />,
            <Quote
              quote={
                Object {
                  "created_date": 2019-01-02T00:00:00.000Z,
                  "display_name": "Teste User 2",
                  "id": "21",
                  "message": "Teste 2",
                  "username": "teste",
                }
              }
            />,
          ],
          <div
            className="col text-center"
          >
            <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Ver mais
            </button>
          </div>,
        ],
        "className": "row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "22",
          "nodeType": "class",
          "props": Object {
            "quote": Object {
              "created_date": 2019-01-01T00:00:00.000Z,
              "display_name": "Teste User",
              "id": "22",
              "message": "Teste",
              "username": "teste",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "21",
          "nodeType": "class",
          "props": Object {
            "quote": Object {
              "created_date": 2019-01-02T00:00:00.000Z,
              "display_name": "Teste User 2",
              "id": "21",
              "message": "Teste 2",
              "username": "teste",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-primary"
              onClick={[Function]}
            >
              Ver mais
            </button>,
            "className": "col text-center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ver mais",
              "className": "btn btn-primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Ver mais",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
